### PR TITLE
Have danger suggest lint fixes

### PR DESCRIPTION
**Pull request by cobbal**
refs: MBL-14001
affects: none
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/dz6hzsq) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fdz6hzsq)
teacher: [Link](https://rebrand.ly/4xuai2g) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F4xuai2g)
parent: [Link](https://rebrand.ly/3z3y0ly) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F3z3y0ly)

**Comment by inst-danger** [comment]
```suggestion
        var endDate: Date?
        var contextCodes: [String] = []
        var filter: String = ""
```

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: None

[MBL-14001](https://instructure.atlassian.net/browse/MBL-14001)

<details><summary>:x: Swift lint</summary>


    ⚠️  /Users/vagrant/git/Core/Core/Planner/GetPlannablesUseCase.swift:27:9: Colon Violation: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. (colon)
    Linting 'PlannablesRequestable.swift' (428/675)
    Linting 'APIPlannable.swift' (429/675)
</details>
<details><summary>:x: XCTest failed: ParentUnitTests/AddStudentControllerTests/testAddPairingCode</summary>


    Crash: Parent (10036): ParentTestCase.router.getter
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Planner/GetPlannablesUseCase.swift [hunk 1]
```diff
@@ -24,7 +24,7 @@ public class GetPlannables: CollectionUseCase {
     var userID: String?
     var startDate: Date?
     var endDate: Date?
-    var contextCodes: [String] = []
+    var contextCodes : [String] = []
     var filter: String = ""
```
Comment (inst-danger) [inline Core/Core/Planner/GetPlannablesUseCase.swift]:
```
suggestion
        var endDate: Date?
        var contextCodes: [String] = []
        var filter: String = ""
```

Comment (inst-danger) [inline Core/Core/Planner/GetPlannablesUseCase.swift]:
```suggestion
        var endDate: Date?
        var contextCodes: [String] = []
        var filter: String = ""
```

Comment (inst-danger) [inline Core/Core/Planner/GetPlannablesUseCase.swift]:
```suggestion
    var endDate: Date?
    var contextCodes: [String] = []
    var filter: String = ""
```

File: Core/Core/Planner/GetPlannablesUseCase.swift [hunk 2]
```diff
@@ -24,7 +24,7 @@ public class GetPlannables: CollectionUseCase {
     var userID: String?
     var startDate: Date?
     var endDate: Date?
-    var contextCodes: [String] = []
+    var contextCodes : [String] = []
```
Comment (inst-danger) [inline Core/Core/Planner/GetPlannablesUseCase.swift]:
```suggestion
    var contextCodes: [String] = []
```

Comment (inst-danger) [inline Core/Core/Planner/GetPlannablesUseCase.swift]:
```suggestion
    var contextCodes: [String] = []
```

File: Core/Core/Planner/APIPlannable.swift [hunk 3]
```diff
@@ -24,8 +24,8 @@ public struct APIPlannable: Codable, Equatable {
     let planner_override: APIPlannerOverride?
     let plannable_id: ID
     let plannable_type: String
-    let html_url: URL?
-    let context_image: URL?
+    let html_url : URL?
+    let context_image : URL?
```
Comment (cobbal) [inline Core/Core/Planner/APIPlannable.swift]:
```suggestion
    let html_url: URL?
    let context_image: URL?
```

File: Core/Core/Grades/Grades.swift [hunk 4]
```diff
@@ -168,7 +168,7 @@ public class Grades {
                         NSPredicate(format: "%K == %@", #keyPath(Enrollment.userID), item.user_id),
                         NSPredicate(format: "%K == %@", #keyPath(Enrollment.roleID), item.role_id),
                         NSPredicate(format: "%K == %@", #keyPath(Enrollment.role), item.role),
-                        NSPredicate(format: "%K == %@", #keyPath(Enrollment.course.id), self.courseID),
+                        NSPredicate(format: "%K == %@", #keyPath(Enrollment.course.id), self.courseID)
```
Comment (inst-danger) [inline Core/Core/Grades/Grades.swift]:
```suggestion
                        NSPredicate(format: "%K == %@", #keyPath(Enrollment.course.id), self.courseID),
```

File: Core/Core/Grades/Grades.swift [hunk 5]
```diff
@@ -258,6 +258,7 @@ public class Grades {
     }
 }
 
+
```
Comment (inst-danger) [inline Core/Core/Grades/Grades.swift]:
```suggestion
```

File: Core/Core/Planner/APIPlannable.swift [hunk 6]
```diff
@@ -24,10 +24,10 @@ public struct APIPlannable: Codable, Equatable {
     let planner_override: APIPlannerOverride?
     let plannable_id: ID
     let plannable_type: String
-    let html_url: URL?
-    let context_image: URL?
+    let html_url : URL?
```
Comment (inst-danger) [inline Core/Core/Planner/APIPlannable.swift]:
```suggestion
    let context_image: URL?
```

File: Core/Core/Planner/APIPlannable.swift [hunk 7]
```diff
@@ -24,10 +24,10 @@ public struct APIPlannable: Codable, Equatable {
...
+    let context_image : URL?
```
Comment (inst-danger) [inline Core/Core/Planner/APIPlannable.swift]:
```suggestion
    let html_url: URL?
```

File: Core/Core/Planner/APIPlannable.swift [hunk 8]
```diff
@@ -24,10 +24,10 @@ public struct APIPlannable: Codable, Equatable {
...
     let context_name: String?
-    let title: String?
+    let title : String?
```
Comment (inst-danger) [inline Core/Core/Planner/APIPlannable.swift]:
```suggestion
    let title: String?
```

File: Core/Core/Planner/GetPlannablesUseCase.swift (hunk 2 above)
Comment (inst-danger) [inline Core/Core/Planner/GetPlannablesUseCase.swift]:
```suggestion
    var contextCodes: [String] = []
```

File: Core/Core/Grades/Grades.swift (hunk 4 above)
Comment (inst-danger) [inline Core/Core/Grades/Grades.swift]:
```suggestion
                        NSPredicate(format: "%K == %@", #keyPath(Enrollment.course.id), self.courseID),
```

File: Core/Core/Grades/Grades.swift (hunk 5 above)
Comment (inst-danger) [inline Core/Core/Grades/Grades.swift]:
```suggestion
```

File: Core/Core/Planner/APIPlannable.swift (hunk 6 above)
Comment (inst-danger) [inline Core/Core/Planner/APIPlannable.swift]:
```suggestion
    let html_url: URL?
```

File: Core/Core/Planner/APIPlannable.swift (hunk 7 above)
Comment (inst-danger) [inline Core/Core/Planner/APIPlannable.swift]:
```suggestion
    let context_image: URL?
```

File: Core/Core/Planner/APIPlannable.swift (hunk 8 above)
Comment (inst-danger) [inline Core/Core/Planner/APIPlannable.swift]:
```suggestion
    let title: String?
```

File: Core/Core/Planner/GetPlannablesUseCase.swift (hunk 2 above)
Comment (inst-danger) [inline Core/Core/Planner/GetPlannablesUseCase.swift]:
```suggestion
    var contextCodes: [String] = []
```

**Comment by cobbal** [comment]
After spending half a day getting this working, I'm not super excited about how it turned out. *shrug* If people want this, it's done and mergable, if not then I'll just close the PR

File: scripts/build_automation/suggest-lint-fix.swift [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+            "-H", "Authorization: Bearer \(token)",
+            "--data-binary", "@-"
+        ).inputJSON(from: GraphQLRequest(
+            query: query,
+            variables: [ "input": input ]
+        )) | cmd("jq")).run()
+    }
+}
+
+func inDir<R>(_ path: String, thunk: () throws -> R) rethrows -> R {
```
Comment (bluwave) [inline scripts/build_automation/suggest-lint-fix.swift]: Just a knit pick, seems like handler or completionHandler fits more with the swift paradigm than thunk

**Comment by bluwave** [comment]
I think it's pretty cool, may be fun to try it and see how useful it is

**Pull request merged**